Cleanup IRC (#4446)
ppy/osu-wiki: wiki/Internet_Relay_Chat/en.md

# Internet Relay Chat

> *For further information, see [Internet Relay Chat (Wikipedia)](https://en.wikipedia.org/wiki/Internet_Relay_Chat "Wikiepdia")*

Reviewer: ```suggestion
*For further information, see [Internet Relay Chat (Wikipedia)](https://en.wikipedia.org/wiki/Internet_Relay_Chat "Wikipedia")*
```
typo

> *For further information, see [Internet Relay Chat (Wikipedia)](https://en.wikipedia.org/wiki/Internet_Relay_Chat "Wikipedia")*

Reviewer 2: ```suggestion
*For further information, see: [Internet Relay Chat (Wikipedia)](https://en.wikipedia.org/wiki/Internet_Relay_Chat "Wikipedia")*
```

> The **Intenet Relay Chat** (**IRC**) is an application layer protocol used for chatting with numerous online clients.

Reviewer 2: ```suggestion
The **Internet Relay Chat** (**IRC**) is an application layer protocol used for chatting with numerous online clients.
```

> The **Intenet Relay Chat** (**IRC**) is an application layer protocol for chatting with numerous clients available to connect with. Although not developed by/for osu!, it is an integral part of its infrastructure and to community [bots](/wiki/Bot_account).

Reviewer 3: this reads really weirdly.

Author: Assuming you mean the ending bit "...and to community bots"?

Reviewer 3: "available to connect with" "Although not developed by/for osu!" "it is an integral part of its infrastructure and to community"

## osu!Bancho

> *Notice to osu!Bancho users: this IRC protocol only a partial implementation, and thus it is not guaranteed that all of a clients' features will work properly on it. (E.g., [HexChat](https://hexchat.github.io/))*

Reviewer 4: ```suggestion
*Notice to osu!Bancho users: this IRC protocol is only a partial implementation, and thus it is not guaranteed that all clients' features will work properly on it. (E.g., [HexChat](https://hexchat.github.io/))*
```
missing word, and the ending reads weird

> *Notice to osu!Bancho users: this IRC protocol is modified, and thus it is not guaranteed that all of a clients' features will work properly on it. (E.g., [HexChat](https://hexchat.github.io/))*

Reviewer 3: rather than "protocol is modified", how about saying it is a naive/partial implementation?

> *osu!Bancho* (sometimes shortened to just *Bancho*) is a modified version of the IRC written by [peppy](https://osu.ppy.sh/users/2) that offers a gateway for in-game chat access; designed to allow users to connect with their own client and chat with other people without the use of the osu! game client.

Reviewer 3: no it's not. bancho is the game server. it has an irc gateway but it's not a "modified version of the IRC". this reads worse than the original

## Connection

> Users wishing to set up a client with osu!Bancho must first set up its sever settings after creating an initial client. The information needed for these settings is listed below:

Reviewer 3: this reads worse than original. "creating an initial client"? "setup a client with osu!bancho"?

Reviewer 3: "sever"?

- Server: `irc.ppy.sh`
- Port: `6667` (default)
- Username: Client creator's osu! username (replace spaces to underscores) (e.g., `beppy master 1000` becomes `beppy_master_1000`)
- Password: (Obtainable at the [IRC Authentication](https://osu.ppy.sh/p/irc) page).

> *Warning: Every users' IRC password is different from their account password. **Do not** share this password with others.*

Reviewer 3: this reads worse than original.

Author: Assuming you mean that it should be "separate" and not "different"? Or just that it sounds clunky?

Reviewer 3: Written in first person sounds better. "their" "this" us weird.

that said there's an issue in the original "to others" should be "with others".

## Disabling join/part messages

> Many IRC clients send join and part (leave) messages for when a player has joined/left a channel by defauly. Although useful to some, it can be irritating to others. Users wishing to disable these messages should refer to their prefered client's documentation. A list of common IRC clients and how to disable these messages is described below:

Reviewer: ```suggestion
Many IRC clients send join and part (leave) messages for when a user has joined/left a channel by default. Although useful to some, it can be irritating to others. Users wishing to disable these messages should refer to their preferred client's documentation. A list of common IRC clients and how to disable these messages is described below:
```

| IRC client | Description |
| :-- | :-- |
| [HexChat](https://hexchat.github.io/ "GitHub") | Check "Hide join and part messages" under Settings - Preferences -> Chatting -> General (Advanced pre-2.9.6)  |
| [ircII](http://www.eterna.com.au/ircii/ "ircII") | Type `IGNORE * CRAP` |
| [Irssi](https://irssi.org "Irssi") | Type `/ignore -channels #somechannel * JOINS PARTS QUITS` |
| [Weechat](https://weechat.org/ "Weechat") | Type `/filter add joinquit * irc_join,irc_part,irc_quit *`. |
| [KVIrc](https://www.kvirc.net/ "KVIrc") | (Refer to [the KVIrc wiki page](https://github.com/kvirc/KVIrc/wiki/FAQ#how-do-i-suppress-join-part-and-quit-messages "GitHub")) |
| [mIRC](https://www.mirc.com/ "mIRC") | Go to mIRC Options (Tools - Options / Alt + O), under IRC tree, click "Events..." button and change Joins and Parts to "Hide" |
| [Quassel IRC](https://quassel-irc.org/ "Quassel IRC") | Right-click inside the chat and choose Joins/Parts/Quits in the "Hide Events" menu. |
| [XChat](http://xchat.org/ "XChat") | Type `/set irc_conf_mode 1` |

## FAQ

### I'm receiving the error "Bad Authentication Token"

> If you're recieving this error, you should first try the following:

Reviewer 2: ```suggestion
If you're receiving this error, you should first try the following:
```

1. Make sure you are using the correct password at the [IRC Authentication page](https://osu.ppy.sh/p/irc).
2. If your username has spaces, replace it to underscores (e.g. **This Username** to **This_Username**)

### Can I use another username?

No. You can only use your osu! username.

### What is this "voice status" that I have?

> Users with *voice status* are also connected using an IRC client (except for chat moderators who always have *operator (+o)* status, regardless of what client they use); users who do not have any status are connected using the game client.

Reviewer: ```suggestion
Users with *voice status* are also connected using an IRC client (except for chat moderators who always have *operator (+o)* status, regardless of what client they use). Users who do not have any status are connected using the game client.
```
I don't think a semicolon is needed here. just separate it with a full stop into two different sentences

Reviewer 4: Also suggest changing 'regardless of what client' to 'regardless of which client' in this line.

Reviewer: `regardless of the client` sounds better but good catch!
